### PR TITLE
Replaced service borg dice with new borg die.

### PR DESCRIPTION
**Pull request by Faptastic**
8 dice squashed into one, the bloat is gone!
I would welcome an improved underlay if you have one, not my forte.

### CODE REVIEW

**Comment by Exxion** [comment]
Now a d00 can't roll below 10
There's a reason it couldn't reach 100 before
Because it goes from 0-90
Which you add to a d10 to get a 1-100 roll

File: code/game/objects/items/weapons/dice.dm [hunk 1]
```diff
@@ -7,12 +7,40 @@
...
+/obj/item/weapon/dice/borg //8 in 1
+	name = "poly-d6"
+	desc = "A polymorphic die made of a mysterious material."
+	var/possible_sides = list(2,4,6,8,10,12,20,100)
+
+/obj/item/weapon/dice/borg/verb/set_die_sides() //so we can change die sides
+	set name = "Set Poly-die Sides"
+	set category = "Object"
+	set src in range(0)
+	set_sides()
```
Comment (PJB3005) [inline code/game/objects/items/weapons/dice.dm]: Gonna need some sanity (is the person alive, etc...)

**Comment by ihadtoregisterforthis** [comment]
Maybe a digital die would be more fitting than one made from a "mysterious material".

**Comment by ComicIronic** [comment]
Digi-die does sound like a good idea.
Also, perhaps get a new icon.

**Comment by Faptastic** [comment]
digidie
This better? I'm not a sprite artist but its hopefully decent enough to look like a small tablet with a picture of a die on it. If someone has a better one I welcome it.

At least this way you can keep the fancy dice roll icons but handwave the fact the die magically changes shape,

**Comment by PJB3005** [comment]
Could you add the change functionality to the dice as am alt click functionality? That'd be a lot quicker than right click.

**Comment by ComicIronic** [comment]
Wait, it's a verb?

DELET

**Comment by PJB3005** [comment]
Issue with removing the verb is that now a lot of people won't be able to figure out how to change the amount of sides.

**Comment by ComicIronic** [comment]
It could be a contextual click on the icon which had some design to encourage you to click it.

**Comment by Faptastic** [comment]
scrnshot1

I've added a contextual click and a little yellow button to go with it, seems to work well enough.
Also tried the device as a human should that ever happen, works exactly like other die.
The description was also modified to mention the button as well as the changelog, I'm sure the player will eventually get the hint.

Unless you have some tweaks to this or suggestions I'm about done.

File: code/game/objects/items/weapons/dice.dm [hunk 2]
```diff
@@ -290,3 +293,54 @@
...
+			return 1
+
+/datum/context_click/digi_die/action(obj/item/used_item, mob/usr, params)
+	var/obj/item/weapon/dice/borg/d = holder
+	if(return_clicked_id_by_params(params))
+		d.set_sides()
+		return 1
+
+/obj/item/weapon/dice/borg/proc/set_sides()
+	var/S = input("Number of sides:") as null|anything in possible_sides
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/dice.dm:333]: This is going to be a sanity issue. Pass in user for this proc and check the user is still next to the dice after the input.

File: code/game/objects/items/weapons/dice.dm [hunk 3]
```diff
@@ -326,11 +326,13 @@
 /datum/context_click/digi_die/action(obj/item/used_item, mob/usr, params)
 	var/obj/item/weapon/dice/borg/d = holder
 	if(return_clicked_id_by_params(params))
-		d.set_sides()
+		d.set_sides(usr)
 		return 1
 
-/obj/item/weapon/dice/borg/proc/set_sides()
+/obj/item/weapon/dice/borg/proc/set_sides(mob/usr)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/dice.dm]: Uh, I'm wondering how this even compiled but make this `user` instead and then use it below.

Comment (Faptastic) [inline code/game/objects/items/weapons/dice.dm]: Do I change the other instances of 'usr' as well that don't cause a compile error when changed? I remember some sort of crusade against either usr or user but I couldn't find which.
I'm testing each revision before it's pushed so thats something at least.

Comment (ComicIronic) [inline code/game/objects/items/weapons/dice.dm]: usr is a special Byond var, and you should almost never be using it. If you're defining your own var, use user.

**Comment by ComicIronic** [comment]
Nicely done on the context clicker, people need to use those more (and not just because I made them)

**Pull request merged**